### ISSUE TITLE
Tracking Issue for `windows_change_time`

### ISSUE THREAD

**Issue by juliusl** [rust-lang/rust#121478]
In Windows, there's a field in FILE_BASIC_INFO called ChangeTime that tracks when file metadata is changed, such as renaming, file attributes, etc.

Currently, this property isn't included in the ::sys::windows::fs::FileAttr implementation, but is available in the c::FILE_BASIC_INFO impl.

Last write time (which is currently provided) is the last time the data stream changed, and this value represents a different time value.

[source](https://devblogs.microsoft.com/oldnewthing/20100709-00/?p=13463#:~:text=I%E2%80%99m%20told%20that%20the%20difference%20is%20metadata.%20The,attributes%20%28hidden%2C%20read-only%2C%20etc.%29%20or%20renaming%20the%20file.)

This change adds a method `change_time()` to the `MetadataExt` trait in order to provide this value.

### Public API

```rust
fn windows_change_time() {
    let meta = std::fs::metadata("example.txt").unwrap();

    // Returns Option<u64> since this is only available if FILE_BASIC_INFO is the backing type of `.inner()`
    let change_time = meta.change_time();
}
```

### Steps / History

- [x] Implementation: #112328
- [ ] Final comment period (FCP)[^1]
- [ ] Stabilization PR

### Unresolved Questions

- None yet.

[^1]: https://std-dev-guide.rust-lang.org/feature-lifecycle/stabilization.html

**Comment by tgross35** [comment]
The docs currently say:

> This will return None if the Metadata instance was not created using the FILE_BASIC_INFO type.

Could you reword this to clarify what is required for somebody using Rust's APIs? It isn't clear how somebody may or may not wind up with a `FILE_BASIC_INFO`, since that type isn't mentioned anywhere.

**Comment by juliusl** [comment]
@tgross35  FILE_BASIC_INFO is a windows api struct and actually seeing the alternative implementation I think we could align both implementations so that I can return a value in both cases, see -- https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getfileinformationbyhandleex#remarks 

I'll tinker and see if I can draft something

**Comment by tgross35** [comment]
> @tgross35 FILE_BASIC_INFO is a windows api struct 

As in, to non-uwp users this only comes up if they are using the `windows` crate? If that is accurate then it would just be good to mention, I just couldn't figure out how this gets used. But...

and actually seeing the alternative implementation I think we could align both implementations so that I can return a value in both cases, see -- https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getfileinformationbyhandleex#remarks
> 
> I'll tinker and see if I can draft something

that sounds even better, cool.
